src/Http/RedirectBinding.php

```diff
@@ -149,14 +150,118 @@ public function processSignedRequest(Request $request)
     }
 
     /**
+     * @param Request $request
+     * @return ReceivedAuthnRequest
+     */
+    public function receiveUnsignedAuthnRequestFrom(Request $request)
```

Comment:
And if needed, a `receiveAuthnRequest` with a simple `if ($query->isSigned())` on whether or not to validate the authnrequest should also be provided for those who want to do both? I mean, you can't know as outside user of this binding whether or not a request is signed without parsing it, which would kinda be against the purpose of this exercise. Arguably, that should be the only method, with two optional configuration parameters: `$allowSignedRequests = true` and `$requestMustBeSigned = true`.